backend/src/v5/models/revisions.js

```diff
@@ -42,43 +45,81 @@ const findOneRevisionByQuery = async (teamspace, model, query, projection, sort)
+	return db.count(teamspace, collectionName(undefined, model), query);
 };
 
-Revisions.getRevisions = (teamspace, model, showVoid, projection = {}) => {
+Revisions.getRevisions = (teamspace, model, modelType, showVoid, projection = {}) => {
 	const query = { ...excludeIncomplete };
 
 	if (!showVoid) {
 		query.void = excludeVoids.void;
 	}
 
-	return findRevisionsByQuery(teamspace, model, query, projection, { timestamp: -1 });
+	return findRevisionsByQuery(teamspace, model, modelType, query, projection, { timestamp: -1 });
 };
 
-Revisions.getRevisionByIdOrTag = (teamspace, model, revision, projection = {}) => findOneRevisionByQuery(teamspace,
-	model, { $or: [{ _id: revision }, { tag: revision }] }, projection);
+Revisions.getRevisionByIdOrTag = (teamspace, model, modelType, revision, projection = {}) => findOneRevisionByQuery(
+	teamspace, model, modelType, { $or: [{ _id: revision }, { tag: revision }] }, projection);
 
-Revisions.updateRevisionStatus = async (teamspace, project, model, revision, status) => {
-	const res = await db.findOneAndUpdate(teamspace, collectionName(model),
+Revisions.addRevision = async (teamspace, project, model, modelType, data) => {
+	const newRevision = {
+		_id: generateUUID(),
+		project,
+		model,
+		timestamp: new Date(),
+		...data,
+	};
+
+	await db.insertOne(teamspace, collectionName(modelType, model), newRevision);
+
+	publish(events.NEW_REVISION, { teamspace, project, model, modelType, revision: newRevision._id });
+
+	return newRevision._id;
+};
+
+Revisions.deleteModelRevisions = (teamspace, project, model, modelType) => db.deleteMany(
+	teamspace, collectionName(modelType, model), { project, model });
+
+Revisions.updateRevision = async (teamspace, model, modelType, revision, updateData) => {
+	const res = await db.findOneAndUpdate(teamspace, collectionName(modelType, model),
 		{ $or: [{ _id: revision }, { tag: revision }] },
-		{ $set: { void: status } },
+		{ $set: updateData },
 		{ projection: { _id: 1 } });
 
 	if (!res) {
 		throw templates.revisionNotFound;
 	}
 
-	publish(events.REVISION_UPDATED, { teamspace, project, model, data: { _id: res._id, void: status } });
+	return res;
+};
+
+Revisions.updateRevisionStatus = async (teamspace, project, model, modelType, revision, status) => {
+	const res = await Revisions.updateRevision(teamspace, model, modelType, revision, { void: status });
+
+	publish(events.REVISION_UPDATED, { teamspace,
+		project,
+		model,
+		modelType,
+		data: { _id: res._id, void: status } });
 };
 
 Revisions.isTagUnique = async (teamspace, model, tag) => {
 	try {
-		await findOneRevisionByQuery(teamspace, model, { tag });
+		await findOneRevisionByQuery(teamspace, model, undefined, { tag });
+		return false;
+	} catch {
+		return true;
+	}
+};
+
+Revisions.isRevAndStatusCodeUnique = async (teamspace, model, revCode, statusCode) => {
+	try {
+		await findOneRevisionByQuery(teamspace, undefined, modelTypes.DRAWING, { revCode, statusCode });
```

Comment:
we need model to be part of the query.

We only want it to be unique within the same drawing